addons/web_editor/static/lib/odoo-editor/src/OdooEditor.js

```diff
@@ -898,6 +898,9 @@ export class OdooEditor extends EventTarget {
      * @see historyUndo
      */
     historyRedo() {
+        this.historyRevert(this._currentStep);
+        this._currentStep.mutations = [];
```

Comment:
@Goaman This code is actually taken from `historyUndo` at https://github.com/odoo/odoo/pull/126301/files#diff-6bf09b72528757673aa5f73488708a44ba1af178d51205248e9bb8429d3e6584L877. We think it was a mistake that those lines were not _also_ in `historyRedo`. I'd rather have your opinion in case there is something that we missed.

@cammarosano You should have copy pasted the comments related to those lines too IMO because otherwise it would be very hard for anyone to understand how those lines make sense there.

Reply:
> // The last step is considered an uncommited draft so always revert it.

When it comes to **undo**, this was a choice.

>  // Clean the last step otherwise if no other step is created after, the
>   // mutations of the revert itself will be added to the same step and
>    // grow exponentially at each undo.

As a consequence of this choice, `this.historyRevert(lastStep)` is done and, _at this point, `lastStep.mutations` contains the current step's mutations plus the ones that revert it, with net effect zero_. Might as well just remove them all and save some memory.  Not clearing it would make the "undo" step that is about to be registed unnecessarily bigger. But I'm not sure what is the relation between this and other undo steps, or the mentioned "exponential" growing at each undo.
I believe a more accurate comment would be the italicized text section above.

When it comes to **redo**, this is _not_ a choice. You can only redo when you are in the state resulting of an undo, that is, redo is the undoing of an undo. Once you add some changes after an undo, you cannot redo anymore (redo... what? how?). Tha's roughly equivalent to having the last history step with state "undefined", which would translate to a redo not being possible.
So either we:
- discard the current step's mutations before doing a redo,
- do not allow redoing.

Mirroring what was made for the undo case, I'd say discarding the current step's mutations would be the most consistent choice.

The fact is that it is rather rare to have uncommited current step mutations when we are about to perform a redo. And I believe that it was assumed that there would not exist such current changes.

So, about copy-pasting the comments from undo: 
- the first one is ok, although it could be rephrased into something like "Current step is considered an uncommited draft, so revert it, otherwise a redo would not be possible." Actually I would rather include this in the commit message rather that having it as comment.
- I don't agree (or fully understand) the second one, and would prefer (again) "At this point, lastStep.mutations contains the current step's mutations plus the ones that revert it, with net effect zero" instead.

What do you think?

Reply:
> @Goaman This code is actually taken from `historyUndo` at https://github.com/odoo/odoo/pull/126301/files#diff-6bf09b72528757673aa5f73488708a44ba1af178d51205248e9bb8429d3e6584L877. We think it was a mistake that those lines were not _also_ in `historyRedo`. I'd rather have your opinion in case there is something that we missed.

I also think it make more sense to revert the current step before `historyRedo`.
The consequence I could foresee is if we tries to redo while being in the middle of an operation (eg. using async) that has not yet called `historyStep`. Although it would also be problematic for redo anyway.

Reply:
@cammarosano I like both: "Current step is considered an uncommited draft, so revert it, otherwise a redo would not be possible." and "At this point, lastStep.mutations contains the current step's mutations plus the ones that revert it, with net effect zero." I think both are providing useful information on a very tricky mechanism. Because of the trickiness of the history steps mechanism, I think it really provides useful value to have both sentences as comments. You can indeed have them in the commit message too.

Reply:
The requested comments were added. 
The commit message was also completely rewritten.

Reply:
@cammarosano Thanks ! 👍  Can I ask you to add steps to reproduce the issue in the commit message, though ? Otherwise it would be complicated for anyone not extremely well acquainted with the editor to get a sense of how they can have mutations in a step after an undo while still being able to do redo. Basically it's using the previous mechanism, we know that, but most other people have no way to know that.

Reply:
I have added the steps to reproduce the issue reported in the task to the commit message.